### PR TITLE
EASYOPAC-1211 - Re-factor easyddb_legend to consider new layout.

### PR DESCRIPTION
**Pull request by bozeak**
https://inlead.atlassian.net/browse/EASYOPAC-1211

### CODE REVIEW

File: css/easyddb_legend.css [hunk 1]
```diff
@@ -55,11 +60,40 @@ div.pane-ding-availability-legend .availability-legend {
...
+.js-search-result--availability-link {
+  border-radius: 5px;
+  padding: 5px 11px;
+  color: white !important;
+  display: inline-block;
+  text-decoration: underline;
+  margin: 2px 0 !important;
+}
+
+.availability.available {
```
Comment (guddo) [inline css/easyddb_legend.css]: Shouldn't you use the existing definitions, instead of creating new ones?

Comment (bozeak) [reply css/easyddb_legend.css]: It is not really possible for the current state of ding_availability.

Comment (guddo) [reply css/easyddb_legend.css]: What do you mean? You can just add `.availability.available,` to L25.

Comment (bozeak) [reply css/easyddb_legend.css]: Yes, you are right. There is no sense to duplicate same definitions.

File: css/easyddb_legend.css [hunk 2]
```diff
@@ -16,19 +14,26 @@
...
+  border-radius: 5px;
 }
-
+.availability.available,
+.availability.available.reservable,
 .availability-legend-item.available img {
   background-color: #528c10; /* Green */
 }
 
+.availability.reservable,
```
Comment (guddo) [inline css/easyddb_legend.css:30]: What about:

- `.availability.processed,`
- `.availability.available.not-reservable,` / `.availability.available.unreservable,`
- `.availability.unavailable.reservable,`

?

File: css/easyddb_legend.css [hunk 3]
```diff
@@ -16,19 +14,26 @@
...
 .availability-legend-item.on-loan img {
   background-color: #f0cb28; /* Yellow */
 }
 
+.availability.unavailable,
+.availability.unreservable,
```
Comment (guddo) [inline css/easyddb_legend.css:37]: Shouldn't this be `.availability.unavailable.not-reservable,` / `.availability.unavailable.unreservable,`?

**Comment by bozeak** [comment]
I was working with effective results which were returned by availability module and didn't met such statuses.

**Comment by guddo** [comment]
What does it mean @bozeak ? You need to look into what the possible status'es is.

**Comment by bozeak** [comment]
I'll check this.

**Comment by bozeak** [comment]
@guddo 
Classes available in latest versions of ding2 - development:
- `pending`
- `processed` - In my opinion we don't have to add any styling for this status, because it's always present
- `available`
- `unavailable`
- `reservable`
- `unreservable`

In 7.x-4.2.1 were also was:
- `not-reservable`
- `not-available-online`
but the functionality part where those classes was added, was modified and for now it's `unreservable`.

**Pull request merged**